astropy/io/misc/pyarrow/csv.py

```diff
@@ -159,12 +159,13 @@ def read_csv(
         include_names, dtypes, null_values, timestamp_parsers
     )
 
-    table_pa = csv.read_csv(
-        input_file,
-        parse_options=parse_options,
-        read_options=read_options,
-        convert_options=convert_options,
-    )
+    with get_readable_fileobj(input_file, encoding="binary") as f:
```

Comment:
Not knowing about the pyarrow internals, I'm wondering if this might have a performance impact for the case of a uncompressed file supplied as the file name (str or Path). As a sanity check, can you measure the read time for a large uncompressed CSV file with current master and your new code?